packages/config/src/config.mts

```diff
@@ -5,16 +5,19 @@ import { applyDefaultValues } from './core.mjs'
 import type { GeneratorConfig, GeneratorConfigWithDefaultValues } from './types.mjs'
 import { validateConfig } from './validation.mjs'
 
-export const writeConfigToFile = async (config: GeneratorConfig) =>
-	writeConfigFile({ ...config, $schema: `https://unpkg.com/typesafe-i18n@${version}/schema/typesafe-i18n.json` })
+export const writeConfigToFile = async (config: GeneratorConfig, configPath: string) =>
+	writeConfigFile(
+		{ ...config, $schema: `https://unpkg.com/typesafe-i18n@${version}/schema/typesafe-i18n.json` },
+		configPath,
+	)
 
 export const doesConfigFileExist = async () => doesPathExist(resolve('.typesafe-i18n.json'))
 
-export const readRawConfig = async () =>
-	(await importFile<GeneratorConfig & { $schema?: string }>(resolve('.typesafe-i18n.json'), false)) || {}
+export const readRawConfig = async (configPath: string) =>
+	(await importFile<GeneratorConfig & { $schema?: string }>(resolve(configPath), false)) || {}
 
-export const readConfig = async (): Promise<GeneratorConfig> => {
-	const generatorConfig = await readRawConfig()
+export const readConfig = async (configPath = '.typesafe-i18n.json'): Promise<GeneratorConfig> => {
```

Comment:
I would not use default values anywhere. This makes it easy to miss a spot where we forgot to pass it.

Reply:
Did you have some ideas, how we can save backward compatibility with previous codebase that have hardcoded path to file, without default values?

Reply:
Ideally it get's passed from the CLI and `.typesafe-i18n.json` is the default value there if -p does not get specified.
Without having a look at the code, I currently don't really know where this function get's called. But the call stack to this function should only be initiated from the CLI or the exporter & importer I think